tests/test_scalyr.py

```diff
@@ -254,42 +264,43 @@ def test_add_log_target(monkeypatch, env, fx_scalyr):
     assert agent.first_run is False
 
 
-@pytest.mark.parametrize('env', ENVS)
-def test_add_log_target_no_src(monkeypatch, env, fx_scalyr):
+def test_add_log_target_no_src(monkeypatch, scalyr_env, fx_scalyr):
     patch_os(monkeypatch)
-    patch_env(monkeypatch, env)
 
     target = fx_scalyr['target']
 
+    isdir = MagicMock(side_effect=[True, True])
+    monkeypatch.setattr('os.path.isdir', isdir)
+
     exists = MagicMock()
```

Comment:
```suggestion
    exists = MagicMock(side_effect=[False])
```

Reply:
Including all of those occurrences below

Reply:
That's a lot:
```
$ ag -c 'MagicMock\(\)'
tests/test_main.py:27
tests/test_appdynamics.py:4
tests/test_base.py:1
tests/test_kube.py:8
tests/test_scalyr.py:19
```

And I don't like this patching at all. It's better to replace it with temp files. But do we want to rewrite it now?

Reply:
Oh, you've just inlined already defined `side_effect`s. That's an easy change